### PR TITLE
Add Common component

### PR DESCRIPTION
**Pull request by 3Mydlo3**
**When merged this pull request will:**
- Add common component
- Add function returning nearest map location
- Add function converting time (in hours) into hh:mm:ss
- Add serverside music logic (with queue)

### CODE REVIEW

File: addons/common/functions/fnc_nearestTown.sqf [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#include "script_component.hpp"
+/*
+ * Author: 3Mydlo3
+ * Function returns nearest map location for given position/unit.
+ *
+ * Arguments:
+ * 0: Unit/position to find nearest map location <UNIT/POSITION>
+ *
+ * Return Value:
+ * 0: Nearest map location <STRING>
```
Comment (veteran29) [inline addons/common/functions/fnc_nearestTown.sqf]: It does not return a string, it's a https://community.bistudio.com/wiki/Location (which is actually more useful)
```suggestion
 * 0: Nearest map location <LOCATION>
```

File: addons/common/functions/fnc_nearestTown.sqf [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+params ["_pos"];
+
+if (_pos isEqualType objNull) then {
+    _pos = getPosATL _pos;
+};
+
+// Get nearest locations to object.
+private _towns = nearestLocations [_pos, ["NameVillage", "NameCity", "NameCityCapital"], 2000];
+// Select nearest location from array.
+_towns select 0;
```
Comment (veteran29) [inline addons/common/functions/fnc_nearestTown.sqf]: Will error out if there is no city in 2km range. No semicolon at return.
```suggestion
_towns param [0, locationNull]
```

File: addons/common/functions/fnc_playMusicServer.sqf [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * Public: No
+ */
+
+params ["_musicClassname"];
+
+// Add to queue
+if GVAR(musicPlaying) exitWith {
+    GVAR(musicQueue) pushBack _musicClassname;
+};
```
Comment (veteran29) [inline addons/common/functions/fnc_playMusicServer.sqf:26]: I don't see `GVAR(musicPlaying)` being set to true outside of EH so it seems that tracks will never be added to queue.
Also I don't see any playMusic being done on server outside of `MusicStop`, so will this EH even be fired?

File: addons/common/functions/fnc_timeFormat.sqf [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+};
+// Get _seconds from _time by removing decimal part after multiplying by 60 (number of seconds in minute).
+private _seconds = floor (_time * 60);
+// Add 0 in front of _seconds if it's less than 10.
+if (_seconds <= 9) then {
+	_seconds = format ["0%1", _seconds];
+};
+
+// Create output.
+format ["%1:%2:%3", _hours, _minutes, _seconds]
```
Comment (veteran29) [inline addons/common/functions/fnc_timeFormat.sqf]: There is a https://community.bistudio.com/wiki/BIS_fnc_timeToString function. If it's shitty/unoptimal then own function makes sense.

**Review by veteran29 — CHANGES_REQUESTED** [review]

File: addons/common/functions/fnc_timeFormat.sqf (hunk 4 above)
Comment (3Mydlo3) [reply addons/common/functions/fnc_timeFormat.sqf]: I guess in this case own function does not make any sense, will be removed. BI function gives the same result.

File: addons/common/functions/fnc_playMusicServer.sqf (hunk 3 above)
Comment (3Mydlo3) [reply addons/common/functions/fnc_playMusicServer.sqf:26]: Yes, I totally forgot to include a way to play music for the first time. I think it is enough to replace `exitWith` with `then else` and use it to raise `playMusic` event.

File: addons/common/functions/fnc_nearestLocation.sqf [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+params ["_pos"];
+
+if (_pos isEqualType objNull) then {
+    _pos = getPosATL _pos;
+};
+
+// Get nearest locations to object.
+private _towns = nearestLocations [_pos, ["NameVillage", "NameCity", "NameCityCapital"], 2000];
+// Select nearest location from array.
+_towns param [0, locationNull];
```
Comment (veteran29) [inline addons/common/functions/fnc_nearestLocation.sqf]: Remove comma on return statement
```suggestion
_towns param [0, locationNull]
```

**Review by veteran29 — CHANGES_REQUESTED** [review]

**Review by veteran29 — APPROVED** [review]

**Pull request merged**